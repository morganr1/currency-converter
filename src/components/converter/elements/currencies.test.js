import React from 'react';
import { shallow } from 'enzyme';
import { Currencies } from './Currencies.element';

describe('Currencies element', () => {
    it('renders correctly with the required props passed to it', () => {
        const mockProps = {
            currencies: ['EUR', 'GBP', 'USD'],
            fromCurrency: 'GBP',
            toCurrency: 'EUR',
            handleCurrencySelect: jest.fn()
        };
        const wrapper = shallow(<Currencies {...mockProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
