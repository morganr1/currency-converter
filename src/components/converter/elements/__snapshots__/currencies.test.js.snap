// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currencies element renders correctly with the required props passed to it 1`] = `
<div
  className="currencies-container"
>
  <DropDownComponent
    currentValue="GBP"
    label="From"
    onChangeCallback={[MockFunction]}
    values={
      Array [
        "EUR",
        "GBP",
        "USD",
      ]
    }
  />
  <DropDownComponent
    currentValue="EUR"
    label="To"
    onChangeCallback={[MockFunction]}
    values={
      Array [
        "EUR",
        "GBP",
        "USD",
      ]
    }
  />
</div>
`;
